pyproject.toml

```diff
@@ -123,21 +119,65 @@ exclude_lines = [
     "if __name__ == .__main__.:",
 ]
 
-[tool.black]
-line-length = 100
-
-[tool.isort]
-profile = "black"
-src_paths = ["src", "tests"]
-
-
 [tool.pyright]
 include = ["src", "tests"]
 
+[tool.ruff]
+src = ["src", "test"]
+line-length = 100
 
-[tool.pylint.'MESSAGES CONTROL']
-disable = "R0913"
-
+[tool.ruff.lint]
+select = [
+    "F",
+    "E",
+    "W",
+    "I",
+    "N",
+    "UP",
+    "YTT",
+    "S",
+    "B",
+    "A",
+    "C4",
+    "DTZ",
+    "EM",
+    "LOG",
+    "G",
+    "PIE",
+    "PT",
+    "RSE",
+    "RET",
+    "SIM",
+    "ARG",
+    "PTH",
+    "PL",
+    "TRY",
+    "PERF",
+    "RUF",
+]
+fixable = [
+    "F401",
+    "F541",
+    "I",
+    "D",
+    "UP",
+    "B",
+    "C4",
+    "EM",
+    "PIE",
+    "PT",
+    "RSE",
+    "RET",
+    "SIM",
+    "PERF",
+    "RUF",
+]
+ignore = [
+    # ignore for compatibility with formatter
+    "W191", "E111", "E114", "E117", "S321",
+    # other
+    "PLR0913",
+]
 
-[tool.pylint.format]
-max-line-length = 100
+[tool.ruff.lint.per-file-ignores]
+"tests/*" = ["S101"]
```

Comment:
Did we also want to ignore src/biocommons/example/tests for S101?

Reply:
I am not sure if tests frequently live within `src/` in the actual repos? Obviously it makes maintaining the examples in this repo easier but it'd be kind of annoying to delete this line when it's applied to repos themselves.

Reply:
Good point